### ISSUE TITLE
My redox-w interferes with my wireless mouse - how do I change channel/frequency for this keyboard/receiver??

### ISSUE THREAD

**Issue by PaterFrog** [mattdibi/redox-w-firmware#13]
Hey there.

I bought my Redox Wireless from Falbatech and they redirected me here to take a look at your firmware. Apparently, Falba uses their own firmware for added range and this seems to cause issues with my mouse. As such, I was wondering if it is enough to update the receiver or the keyboard with your firmware, or if I need to make any changes to it.

I am using QMK to program the keyboard layout and flash with the QMK Toolbox, so I personally have very little idea how to go about updating the firmware manually, or even which I need to update.

Do you need more information?

**Comment by mattdibi** [comment]
Hi there,

this is a fairly unfortunate situation.

I would suggest you to try and modify the channel table for the keyboard transmitters and receiver.
The part you need to change are [this ](https://github.com/mattdibi/redox-w-firmware/blob/24b9af3e0e53acad5d21891aa4800ba55c17c8b9/redox-w-keyboard-basic/main.c#L45)and [this](https://github.com/mattdibi/redox-w-firmware/blob/24b9af3e0e53acad5d21891aa4800ba55c17c8b9/redox-w-receiver-basic/main.c#L56).

Keep in mind that to burn the new firmware onto the YJ-14015 you'll need a STLinkV2 debugger. [Here](https://github.com/mattdibi/redox-keyboard/tree/master/redox-w#firmware) you can find details on how to compile and upload the firmware to the YJ-14015.

Hope this helps.

**Comment by PaterFrog** [comment]
Hey there, thanks for getting back to me. I ended up resolving the mouse situation by getting a wired mouse instead.

However,  my left keyboard      keeeps repeating lettttttttttttttttttters freqqqqqqqquently. This has always happened, but        only so little that I didn't mind much. Now it's basically constantly.        Do you haaaaaaave any idea what might be causing this? The right side is perfectly fine.
I have the wifi   dongle            almost touching both halvessssssssss, so that should not be the    issssssue, unlessssssss there is hardddware defect with thee sender of   the left half.     Turning it off and on does not help^^.
The interaction between left and righttt is not entirely okay either. I"m typinggg  e and k  alternatively, at maybe 5 strrrrokes  per second:
ekekekkekekekkekekekekekekekekkekekekekekkekkkekekekkekekekkkekekekekekekekekekkkekkekekekekek
You see that   the right side is  uneven, and that is one of the better examples.

Do you figure this is hardware  related, or software? I've ordered an STLinkV22 debugger, it should arrive in a few days.

**Comment by mattdibi** [comment]
> However, my left keyboard keeeps repeating lettttttttttttttttttters freqqqqqqqquently. This has always happened, but only so little that I didn't mind much. Now it's basically constantly. Do you haaaaaaave any idea what might be causing this? The right side is perfectly fine.

If the issue got worse over time it might be due to the battery voltage getting low. We talked about it in this issue here in more detail: https://github.com/mattdibi/redox-keyboard/issues/55

Are you using a steel plate case from Falbatech?

**Comment by PaterFrog** [comment]
Hey. Switching the batterie has _almost_ entirely fixed it. I still get the occasional double letter, and I figure this might be hardware related. But it's very rare (once or twice a day), so I won't bother fixing that.

I use the bamboo case.

**Comment by 66f94eae** [comment]
> Hey there, thanks for getting back to me. I ended up resolving the mouse situation by getting a wired mouse instead.
> 
> However, my left keyboard keeeps repeating lettttttttttttttttttters freqqqqqqqquently. This has always happened, but only so little that I didn't mind much. Now it's basically constantly. Do you haaaaaaave any idea what might be causing this? The right side is perfectly fine. I have the wifi dongle almost touching both halvessssssssss, so that should not be the issssssue, unlessssssss there is hardddware defect with thee sender of the left half. Turning it off and on does not help^^. The interaction between left and righttt is not entirely okay either. I"m typinggg e and k alternatively, at maybe 5 strrrrokes per second: ekekekkekekekkekekekekekekekekkekekekekekkekkkekekekkekekekkkekekekekekekekekekkkekkekekekekek You see that the right side triggers unevenly, and that is one of the better examples.
> 
> Do you figure this is hardware related, or software? I've ordered an STLinkV22 debugger, it should arrive in a few days.
> 
> Slightly more info: Removing the battery for a second helps. Merely turning the keyboard on and off, does not. The repetition is still there, but it's so little that I might get an extra R once or twice per sentence, not the shitshow above. Further, I bought this keyboard only a few months ago, so the battery cell should be fine. I'll throw a new one in to test it though.
> 
> Edit: Switching the battery between left and right has not entirely fixed the issue either. The left is as fine as ever, and on the right, I still occasionally get double letters, though rarely. I still had a repeating left KB letter typing this, but only the one.

yes, my right keyboard is the same with you. I try to change new battery but ....

I notice this [code](https://github.com/mattdibi/redox-w-firmware/blob/master/redox-w-keyboard-basic/main.c#L27), change it to bigger(32), now my keyboard in working oreder
